libr/core/windows_heap.c

```diff
@@ -60,14 +62,42 @@
 	}
 
 #define UPDATE_FLAGS(hb, flags)\
-	if ((flags & 0xf1) || (flags & 0x0200)) {\
+	if (((flags) & 0xf1) || ((flags) & 0x0200)) {\
 		hb->dwFlags = LF32_FIXED;\
-	} else if ((flags & 0x20)) {\
+	} else if ((flags) & 0x20) {\
 		hb->dwFlags = LF32_MOVEABLE;\
-	} else if ((flags & 0x0100)) {\
+	} else if ((flags) & 0x0100) {\
 		hb->dwFlags = LF32_FREE;\
 	}\
-	hb->dwFlags |= (flags >> SHIFT) << SHIFT;\
+	hb->dwFlags |= ((flags) >> SHIFT) << SHIFT;
+
+#define FILL_TYPE(type, flags)\
```

Comment:
Make this into a function. Macros shouldnt be that large if u want to make code properly maintainable